src/Features/CSharp/Portable/ConvertAutoPropertyToFullProperty/CSharpConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,226 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.ConvertAutoPropertyToFullProperty;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Simplification;
+using static Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles.SymbolSpecification;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertAutoPropertyToFullProperty
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(CSharpConvertAutoPropertyToFullPropertyCodeRefactoringProvider)), Shared]
+    internal class CSharpConvertAutoPropertyToFullPropertyCodeRefactoringProvider : AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider
+    {
+        internal override SyntaxNode GetProperty(SyntaxToken token)
+        {
+            var containingProperty = token.Parent.FirstAncestorOrSelf<PropertyDeclarationSyntax>();
+            if (containingProperty == null)
+            {
+                return null;
+            }
+
+            var start = containingProperty.AttributeLists.Count > 0
+                ? containingProperty.AttributeLists.Last().GetLastToken().GetNextToken().SpanStart
+                : containingProperty.SpanStart;
+
+            // Offer this refactoring anywhere in the signature of the property
+            var position = token.SpanStart;
+            if (position < start || position > containingProperty.Identifier.Span.End)
```

Comment:
this doesn't feel quite right.  Have we ensured that hte user's original position is within the token that we're looking at?  otherwise, the user's positino could be in the preceding trivia and we woudl count it as within the header here.